packages/p2p/src/components/buy-sell/popup.jsx

```diff
@@ -17,12 +17,90 @@ import { localize }    from 'Components/i18next';
 import { requestWS }   from 'Utils/websocket';
 import FormError       from '../form/error.jsx';
 
+class SendField extends Component {
+    render() {
+        const {
+            error,
+            handleChange,
+            is_buy,
+            send_currency,
+            setFieldValue,
+        } = this.props;
+        return (
+            <Field name='send'>
+                {({ field }) => (
+                    <Input
+                        {...field}
+                        data-lpignore='true'
+                        type='number'
+                        error={error}
+                        label={localize('Send')}
+                        className='buy-sell__popup-field'
+                        placeholder={localize('Send amount')}
+                        trailing_icon={<span className='buy-sell__popup-field--trailing'>{send_currency}</span>}
+                        onChange={(e) => {
+                            const send = isNaN(e.target.value) ? 0 : e.target.value;
+                            const receive_amount =
+                                this.calculateReceiveAmount(send, is_buy);
+
+                            setFieldValue('receive', receive_amount);
+                            handleChange(e);
+                        }}
+                        required
+                    />
+                )}
+            </Field>
+        );
+    }
+}
+class ReceiveField extends Component {
```

Comment:
same here

Reply:
for consistency? to match the popup?

Reply:
I mean I thought we were using hooks in p2p (+ it's not using any of the life cycle methods), is there any reason for it being a class component instead of a function component?